templates/pre-receive.erb

```diff
@@ -0,0 +1,19 @@
+#! /bin/sh
+while read oldrev newrev refname; do
+  author=$(git log -1 --format='%an' $newrev)
+
+  # superuser always gets a pass
+  if [[ "${author}" == "<%= @admin_username %>" ]]; then
+    continue
+
+  # creating a new branch
+  elif [[ "${oldrev}" == "0000000000000000000000000000000000000000" ]]; then
```

Comment:
@binford2k Is there any protection here against someone creating a feature branch off of someone else's branch instead of their own, or does that matter? I suppose it will be caught later if they try to commit to it anyway?